### PR TITLE
Sof 2364/able to display panel configurations

### PR DESCRIPTION
**Pull request by LindvedKrvang**
Note: Is dependent on: https://github.com/tv2/sofie-tv2-server/pull/64

The Panel Settings page now show a list of all PanelConfigurations in the system. This list is updated when a new Panel is created, updated or deleted.
image

Also in this PR:
- State service for `PanelConfiguration` that listens for events from the backend
- State service for `PanelLayoutConfiguration` that listens for events from the backend. This was need to get the name of the associated PanelLayoutConfiguration for a PanelConfiguration. Note: `inputConfigurations` has **not** been added in this PR as it went too far beyond the scope.

### CODE REVIEW

File: src/app/settings/panel-settings/models/panel-layout-configuration.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { PanelModel, PanelType } from './panel-enums'
+
+export interface PanelLayoutConfiguration {
+  id: string
+  name: string
+  type: PanelType
+  model: PanelModel
+  // TODO: Missing inputConfigurations
```
Comment (MadsThy) [inline src/app/settings/panel-settings/models/panel-layout-configuration.ts:8]:
```suggestion
  // TODO: Missing inputConfigurations
```

Did you forget to remove this? 😄 or is it coming in another PR/SOF-issue?

File: src/app/settings/panel-settings/services/panel-configuration-parser.service.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import { Injectable } from '@angular/core'
+import * as zod from 'zod'
+import { PanelType, SkaarhojModel } from '../models/panel-enums'
+import { PanelConfiguration } from '../models/panel-configuration'
+
+@Injectable()
+export class PanelConfigurationParser {
+  constructor() {}
```
Comment (MadsThy) [inline src/app/settings/panel-settings/services/panel-configuration-parser.service.ts:8]: Empty constructor :)

File: src/app/settings/panel-settings/components/panel-settings-page/panel-settings-page.component.html [hunk 3]
```diff
@@ -1 +1,3 @@
-<p>panel-settings-page works!</p>
+<sofie-page i18n-pageTitle pageTitle="settings.panels.title">
+    <sofie-panel-configurations></sofie-panel-configurations>
```
Comment (MadsThy) [inline src/app/settings/panel-settings/components/panel-settings-page/panel-settings-page.component.html:2]:
```suggestion
    <sofie-panel-configurations />
```

File: src/app/settings/panel-settings/services/panel-layout-configuration-parser.service.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import { Injectable } from '@angular/core'
+import * as zod from 'zod'
+import { PanelType, SkaarhojModel } from '../models/panel-enums'
+import { PanelLayoutConfiguration } from '../models/panel-layout-configuration'
+
+@Injectable()
+export class PanelLayoutConfigurationParser {
+  constructor() {}
```
Comment (MadsThy) [inline src/app/settings/panel-settings/services/panel-layout-configuration-parser.service.ts:8]: empty constructor :)

**Review by MadsThy — APPROVED** [review]

File: src/app/settings/panel-settings/models/panel-layout-configuration.ts (hunk 1 above)
Comment (LindvedKrvang) [reply src/app/settings/panel-settings/models/panel-layout-configuration.ts:8]: It's intentional. It's even mentioned in the PR description:
image

**Review by RasmusAlbrektsen — APPROVED** [review]

**Pull request merged**